pages/api/storefronts.ts

```diff
@@ -1,26 +1,142 @@
-import type { NextApiRequest, NextApiResponse } from 'next'
-import type { Storefront } from '@/modules/storefront/types'
-import ArweaveSDK from '@/modules/arweave/client'
-import { initArweave } from '@/modules/arweave'
+import type { NextApiRequest, NextApiResponse } from 'next';
+import type { Storefront } from '@/modules/storefront/types';
+import ArweaveSDK from '@/modules/arweave/client';
+import { initArweave } from '@/modules/arweave';
+import { ApiError } from '@/modules/utils';
+import Ajv, { JTDSchemaType } from 'ajv/dist/jtd';
+import { ArweaveFile } from '@/modules/arweave/types';
+import { PutStorefrontParams } from '@/modules/storefront/put-storefront';
+import { unpackNotarized, verifyNaclSelfContained } from '@/modules/notary';
+import { PublicKey } from '@solana/web3.js';
+import { WALLETS } from '@/modules/wallet/server';
+import { stylesheet } from '@/modules/theme';
+
+/** JSON schemas for parsing request parameters. */
+const SCHEMAS = (() => {
```

Comment:
Schema is another item we could register on the next config. Really anything we are managing with constant closures like this.